### PR TITLE
Added the variable modal and component and implemented basic renaming…

### PR DESCRIPTION
**Pull request by CoryDCode**
… functionality.

---
This change is [Reviewable](https://reviewable.io/reviews/google/blockly/991)

### CODE REVIEW

File: accessible/variable-modal.service.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/**
+ * AccessibleBlockly
+ *
+ * Copyright 2016 Google Inc.
```
Comment (rachel-fenichel) [inline accessible/variable-modal.service.js]: Fix copyright date

File: accessible/variable-modal.service.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Angular2 Service for the toolbox modal.
```
Comment (rachel-fenichel) [inline accessible/variable-modal.service.js]: Fix fileoverview and author

File: accessible/variable-modal.service.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ */
+
+blocklyApp.VariableModalService = ng.core.Class({
+  constructor: [
+    function() {
+      this.modalIsShown = false;
+    }
+  ],
+  registerPreShowHook: function(preShowHook) {
+    var that = this;
```
Comment (rachel-fenichel) [inline accessible/variable-modal.service.js]: Don't need that = this

File: accessible/variable-modal.component.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+          setTimeout(function() {
+            document.getElementById('mainFieldId').focus();
+          }, 150);
+        }
+      );
+    }
+  ],
+  // Caches the current text variable as the user types.
+  setTextValue: function(newValue) {
+    this.variableName = newValue
```
Comment (rachel-fenichel) [inline accessible/variable-modal.component.js]: missing semicolon

**Comment by RoboErikG** [comment]
Review status: 0 of 6 files reviewed at latest revision, 9 unresolved discussions.

---

*[accessible/field-segment.component.js, line 90 at r1](https://reviewable.io:443/reviews/google/blockly/991#-KfwtXQ9ZVCrrznueA__:-KfwtXQ9ZVCrrznueA_a:b6blpnh) ([raw file](https://github.com/google/blockly/blob/256cecbc7966ebb3c3361fcdf034e7cb010f6591/accessible/field-segment.component.js#L90)):*
> ```JavaScript
>     }
>   },
>   shouldBreakCache: function() {
> ```

please add comments to new methods. The old ones also need comments, but that's outside the scope of this CL. ;)

---

*[accessible/field-segment.component.js, line 135 at r1](https://reviewable.io:443/reviews/google/blockly/991#-KfwunszCZU8KBK7dEo6:-Kfwunt-Jw03k8RRuG1Q:b-7ropln) ([raw file](https://github.com/google/blockly/blob/256cecbc7966ebb3c3361fcdf034e7cb010f6591/accessible/field-segment.component.js#L135)):*
> ```JavaScript
>     this.mainField.setValue(newValue || 0);
>   },
>   selectOption: function() {
> ```

ditto

---

*[accessible/variable-modal.component.js, line 1 at r1](https://reviewable.io:443/reviews/google/blockly/991#-Kfwuu54ZAJTDtWmtnRK:-Kfwuu54ZAJTDtWmtnRL:b-uh4zff) ([raw file](https://github.com/google/blockly/blob/256cecbc7966ebb3c3361fcdf034e7cb010f6591/accessible/variable-modal.component.js#L1)):*
> ```JavaScript
> 
> ```

Missing copyright header

---

*[accessible/variable-modal.component.js, line 118 at r1](https://reviewable.io:443/reviews/google/blockly/991#-Kfwv86S4ey67ckTxt6t:-Kfwv86S4ey67ckTxt6u:bkejl5g) ([raw file](https://github.com/google/blockly/blob/256cecbc7966ebb3c3361fcdf034e7cb010f6591/accessible/variable-modal.component.js#L118)):*
> ```JavaScript
>   // Gets all the interactive elements for the modal.
>   getInteractiveElements: function() {
>     return Array.prototype.filter.call(document.getElementById("varForm").elements,  function(element) {
> ```

nit >100 cols

---

*[accessible/variable-modal.service.js, line 32 at r1](https://reviewable.io:443/reviews/google/blockly/991#-KfwvEil8_CUCNuX4dwS:-KfwvEil8_CUCNuX4dwT:bcmmkyh) ([raw file](https://github.com/google/blockly/blob/256cecbc7966ebb3c3361fcdf034e7cb010f6591/accessible/variable-modal.service.js#L32)):*
> ```JavaScript
>     }
>   ],
>   registerPreShowHook: function(preShowHook) {
> ```

comment new functions.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/991)*

**Pull request merged**